pkg/apis/rollouts/v1alpha1/experiment_types.go

```diff
@@ -186,11 +186,16 @@ type ExperimentList struct {
 	Items []Experiment `json:"items"`
 }
 
+// One of TemplateName or ClusterTemplateName must be specified
 type ExperimentAnalysisTemplateRef struct {
 	// Name is the name of the analysis
 	Name string `json:"name"`
 	// TemplateName reference of the AnalysisTemplate name used by the Experiment to create the run
-	TemplateName string `json:"templateName"`
+	// +optional
+	TemplateName string `json:"templateName,omitempty"`
```

Comment:
Is omitempty on both this and ClusterTemplateName OK for now or do we need validation that exactly one exists?

Reply:
I think so but I'm not sure where to add this.

Reply:
In workflows, we took a different approach and introduced a `clusterScope: true` boolean.

```yaml
  strategy:
    canary:
      analysis:
        templateName: success-rate
        clusterScope: true
```

In this way, templateName was never optional. I'm thinking we should make templateName mandatory with a separate boolean to indicate its a ClusterAnalysisTemplate.

Reply:
Ok, I will make the changes

Reply:
I have made the changes

Reply:
Awesome! if you remove the omitempty from the TemplateName json tag, it should be good to go!